File: data/transcoder_evaluation_gfg/go/PROGRAM_CALCULATE_VOLUME_ELLIPSOID.go
package main

import (
	"math"
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(r1 float32, r2 float32, r3 float32) float32 {
	var pi float32 = 3.14
	return float32(float64(pi) * 1.33 * float64(r1) * float64(r2) * float64(r3))
}
//TOFILL
func main() {
	var (
		n_success int         = 0
		param0    []float32 = []float32{3287.4842316041018, -3707.427510963942, 8980.643174783816, -2698.0187368852694, 8627.156664162168, -7316.329924623669, 7857.3846206400485, -6502.657905007728, 4468.400513325576, -7231.864791620428}
		param1    []float32 = []float32{4503.332888443404, -6671.335781753231, 3584.781688607942, -1004.7289573934537, 9572.27618966978, -6591.043206581106, 3671.761679299217, -1412.2240121470609, 2272.1999139470304, -8036.087711033032}
		param2    []float32 = []float32{8590.24729914204, -2780.4954870801926, 2818.469507143102, -9602.530725071243, 4783.930377855004, -9760.465488363216, 2534.5825334137794, -6135.238350044512, 4753.075799180736, -6456.263512521035}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if float64(math.Abs(float64(1-(float64(math.Abs(float64(f_gold(param0[i], param1[i], param2[i]))))+1e-07)/(float64(math.Abs(float64(f_filled(param0[i], param1[i], param2[i]))))+1e-07)))) < 0.001 {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/c/PROGRAM_CALCULATE_VOLUME_ELLIPSOID.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

float f_gold ( float r1, float r2, float r3 ) {
  float pi = 3.14;
  return 1.33 * pi * r1 * r2 * r3;
}


float f_filled ( float r1, float r2, float r3 ) {}

int main(void) {
    int n_success = 0;
    float param0[] = {3287.4842316041018F,-3707.427510963942F,8980.643174783816F,-2698.0187368852694F,8627.156664162168F,-7316.329924623669F,7857.3846206400485F,-6502.657905007728F,4468.400513325576F,-7231.864791620428F};
    float param1[] = {4503.332888443404F,-6671.335781753231F,3584.781688607942F,-1004.7289573934537F,9572.27618966978F,-6591.043206581106F,3671.761679299217F,-1412.2240121470609F,2272.1999139470304F,-8036.087711033032F};
    float param2[] = {8590.24729914204F,-2780.4954870801926F,2818.469507143102F,-9602.530725071243F,4783.930377855004F,-9760.465488363216F,2534.5825334137794F,-6135.238350044512F,4753.075799180736F,-6456.263512521035F};
    for(int i = 0; i < len(param0); ++i)
    {
        if(abs(1 - (0.0000001 + abs(f_gold(param0[i],param1[i],param2[i])) )/ (abs(f_filled(param0[i],param1[i],param2[i])) + 0.0000001)) < 0.001F)
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}